.github/workflows/pr_checks.yaml

```diff
@@ -0,0 +1,120 @@
+name: PR Checks
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches:
+      - "**"
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Set up Environment
+        id: setup
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv venv
+          echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
+        continue-on-error: false
+
+      - name: Install Package
+        id: install_package
+        run: |
+          . .venv/bin/activate
+          uv pip install "lm_buddy[dev] @ ."
+        continue-on-error: false
+
+      - name: Unit Tests
+        id: unit_tests
+        run: |
+          . .venv/bin/activate
+          pytest tests/unit
+        continue-on-error: false
+
+      - name: Integration Tests
+        id: integration_tests
+        run: |
+          . .venv/bin/activate
+          pytest tests/integration
+        continue-on-error: false
+
+  linting:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Set up Environment
+        id: setup
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv venv
+          echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
+        continue-on-error: false
+
+      - name: Install Dependencies
+        id: install_dependencies
+        run: |
+          . .venv/bin/activate
+          uv pip install toml-cli
+          uv pip install $(toml get --toml-path pyproject.toml "project.optional-dependencies.ruff[0]")
+        continue-on-error: false
+          
+      - name: Linting with Ruff
+        id: ruff
+        run: |
+          . .venv/bin/activate
+          ruff --output-format=github .
+        continue-on-error: false
+
+  version:
+    name: Version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Set up Environment
+        id: setup
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          uv venv
+          echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
+        continue-on-error: false
+
+      - name: Install Dependencies
+        id: install_dependencies
+        run: |
+          . .venv/bin/activate
+          uv pip install toml-cli packaging
+        continue-on-error: false
+
+      - name: Validate Version Bump
```

Comment:
could we use something like this instead since it already exists or are there features that are missing? https://github.com/marketplace/actions/require-semantic-versioning-bump

Reply:
Good question, I didn't know that exists. Let me check later if it will work.